source/src_pw/hamilt_pw.cuh

```diff
@@ -49,28 +55,57 @@ class Hamilt_PW
                   const ModuleBase::ComplexMatrix &psi,
                   ModuleBase::ComplexMatrix &evc,
                   double *en);
-
-    void h_1psi_gpu(
+    
+    void diagH_subspace_cuda(
+        const int ik,
+        const int nstart,
+        const int n_band,
+        const double2 *psi_c, // matrix
+        double2 *evc, // matrix
+        double *en,
+        double2 *vkb_c);
+
+    void h_1psi_cuda(
+        const int npw,
+        const float2 *psi1d,
+        float2 *hpsi,
+        float2 *spsi,
+        float2 *vkb_c);
+    
+    void h_1psi_cuda(
         const int npw,
-        const CUFFT_COMPLEX *psi1d,
-        CUFFT_COMPLEX *hpsi,
-        CUFFT_COMPLEX *spsi);
+        const double2 *psi1d,
+        double2 *hpsi,
+        double2 *spsi,
+        double2 *vkb_c);
 
     void h_1psi(
         const int npw,
         const std::complex<double> *psi1d,
         std::complex<double> *hpsi,
         std::complex<double> *spsi);
 
-    void h_psi_gpu(
-		const CUFFT_COMPLEX *psi,
-		CUFFT_COMPLEX *hpsi,
-		const int m = 1); // qianrui add a default parameter 2021-3-31
-
-    void s_1psi_gpu(
+    void h_psi_cuda(
+		const float2 *psi,
+		float2 *hpsi,
+        float2 *vkb_c,
```

Comment:
Please using tab but not whitespace for indentation. Reformatting would be appreciated.